packages/components/src/tab-panel/test/index.tsx

```diff
@@ -93,193 +93,298 @@ describe( 'TabPanel', () => {
+					children={ () => undefined }
+				/>
+			);
+			expect( getSelectedTab() ).toHaveTextContent( 'Beta' );
+		} );
 
-		expect( screen.getByRole( 'tab', { name: 'Delta' } ) ).toHaveAttribute(
-			'aria-disabled',
-			'true'
-		);
+		it( 'should re-render with a tab selected by prop tabName, when initialTabName not set', async () => {
+			const mockOnSelect = jest.fn();
+			const { rerender } = render(
+				<TabPanel tabs={ TABS } children={ () => undefined } />
+			);
 
-		// onSelect gets called on the initial render.
-		expect( mockOnSelect ).toHaveBeenCalledTimes( 1 );
+			rerender(
+				<TabPanel
+					tabName="beta"
+					tabs={ TABS }
+					children={ () => undefined }
+					onSelect={ mockOnSelect }
+				/>
+			);
 
-		// onSelect should not be called since the disabled tab is highlighted, but not selected.
-		await user.keyboard( '[ArrowLeft]' );
-		expect( mockOnSelect ).toHaveBeenCalledTimes( 1 );
-	} );
+			expect( getSelectedTab() ).toHaveTextContent( 'Beta' );
+			expect( mockOnSelect ).not.toHaveBeenCalled();
+		} );
 
-	it( 'should select the first enabled tab when the inital tab is disabled', () => {
-		const mockOnSelect = jest.fn();
+		it( 'should re-render with a tab selected by prop tabName, when initialTabName is set', () => {
+			const mockOnSelect = jest.fn();
+			const { rerender } = render(
+				<TabPanel
+					initialTabName="beta"
+					tabs={ TABS }
+					children={ () => undefined }
+				/>
+			);
 
-		render(
-			<TabPanel
-				tabs={ [
-					{
-						name: 'alpha',
-						title: 'Alpha',
-						className: 'alpha-class',
-						disabled: true,
-					},
-					{
-						name: 'beta',
-						title: 'Beta',
-						className: 'beta-class',
-					},
-				] }
-				initialTabName="alpha"
-				children={ () => undefined }
-				onSelect={ mockOnSelect }
-			/>
-		);
+			rerender(
+				<TabPanel
+					tabName="gamma"
+					tabs={ TABS }
+					children={ () => undefined }
+					onSelect={ mockOnSelect }
+				/>
+			);
 
-		expect( getSelectedTab() ).toHaveTextContent( 'Beta' );
-	} );
+			expect( getSelectedTab() ).toHaveTextContent( 'Gamma' );
+			expect( mockOnSelect ).not.toHaveBeenCalled();
+		} );
 
-	it( 'should select the first enabled tab when the currently selected becomes disabled', () => {
-		const mockOnSelect = jest.fn();
+		it( 'should fallback to first enabled tab, when active tab is removed', () => {
+			const mockOnSelect = jest.fn();
+			const { rerender } = render(
+				<TabPanel
+					tabName="beta"
+					tabs={ TABS }
+					children={ () => undefined }
+				/>
+			);
 
-		const { rerender } = render(
-			<TabPanel
-				tabs={ TABS }
-				children={ () => undefined }
-				onSelect={ mockOnSelect }
-			/>
-		);
+			rerender(
+				<TabPanel
+					tabs={ TABS.slice( 2 ) /* remove alpha, beta */ }
+					children={ () => undefined }
+					onSelect={ mockOnSelect }
+				/>
+			);
+			expect( getSelectedTab() ).toHaveTextContent( 'Gamma' );
+			expect( mockOnSelect ).toHaveBeenLastCalledWith( 'gamma' );
```

Comment:
`onSelect` isn't called in this scenario because it is a `non-user` interaction. but it doesn't feel right.